utils/utils.go

```diff
@@ -256,3 +257,34 @@ func ValidateRawGithubUrl(url string) error {
 
 	return nil
 }
+
+func TypedErr(e interface{}) error {
+	switch t := e.(type) {
+	case error:
+		return t
+	case string:
+		return errors.New(t)
+	default:
+		return nil
+	}
+}
+
+func WrapError(mainErr interface{}, subErr interface{}) error {
+	var main, sub error
+	main = TypedErr(mainErr)
+	sub = TypedErr(subErr)
+	// Some times the wrap will wrap a nil error
+	if main == nil && sub == nil {
+		return nil
+	}
+
+	if main == nil && sub != nil {
+		return fmt.Errorf("sub error: %w", sub)
+	}
+
+	if main != nil && sub == nil {
+		return fmt.Errorf("%w: unknown sub error", main)
```

Comment:
how about just returning main instead?
this follows the [errors.Join](https://pkg.go.dev/errors#Join) semantic where they just discard any nil error argument

Reply:
Okie, make sense. I was just following the initial returned value style. Changing now.